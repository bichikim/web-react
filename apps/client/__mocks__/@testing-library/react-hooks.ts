export * from '@testing-library/react-hooks'
