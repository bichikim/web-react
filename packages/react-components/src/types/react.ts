import {ReactElement} from 'react'
export type FunctionChildren<Args extends any[] = any[]> = (...args: Args) => ReactElement
