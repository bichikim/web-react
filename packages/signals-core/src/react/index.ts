export * from './deps'
export * from './is-in-component'
export * from './signals'
export * from 'src/types'
