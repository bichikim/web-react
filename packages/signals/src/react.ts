export * from './create'
export * from './types'
export * from '@winter-love/signals-rebuild/react'
