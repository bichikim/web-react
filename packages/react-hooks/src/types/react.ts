/* eslint-disable @typescript-eslint/ban-types */
import type {PureObject} from 'src/utils'

export type ReactComponent<Props extends PureObject> =
  | import('react').ComponentType<Props>
  | import('react').ForwardRefExoticComponent<Props>

export type FC<Props = {}> = import('react').FC<import('react').PropsWithChildren<Props>>
export type FCC<Props = {}> = import('react').FC<
  import('react').PropsWithChildren<Props & {className?: string}>
>
export type FFC<Props = {}, ChildrenProps = {}> = import('react').FC<
  Props & {children?: (props: ChildrenProps) => import('react').ReactNode}
>
export type FPC<Props = {}> = import('react').FC<Props>
