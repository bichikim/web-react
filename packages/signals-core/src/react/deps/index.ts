import {createDeps} from 'src/create'
import {isInComponent} from 'src/react/is-in-component'
import {useMemo, useRef} from 'react'

export const deps = createDeps({
  isInComponent,
  useMemo,
  useRef,
})
