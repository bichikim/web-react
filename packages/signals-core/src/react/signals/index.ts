import {createSignals} from 'src/create'
import {isInComponent} from 'src/react/is-in-component'
import {useMemo, useRef} from 'react'

export const signals = createSignals({
  isInComponent,
  useMemo,
  useRef,
})

export const useSignals = signals
