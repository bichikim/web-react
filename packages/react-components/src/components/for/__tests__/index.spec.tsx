/**
 * @jest-environment jsdom
 */
import {For} from '../'
import {render, screen} from '@testing-library/react'

describe('For', () => {
  it('should render list all', async () => {
    render(<For each={[1, 2, 3, 4, 5]}>{(item) => <div data-testid="item">{item}</div>}</For>)

    expect(await screen.findAllByTestId('item')).toMatchSnapshot()
  })
  it('should render fallback', async () => {
    render(
      <For each={[]} fallback={<div data-testid="loading">Loading...</div>}>
        {(item) => <div data-testid="item">{item}</div>}
      </For>,
    )

    expect(await screen.findAllByTestId('loading')).toMatchSnapshot()
  })
})
