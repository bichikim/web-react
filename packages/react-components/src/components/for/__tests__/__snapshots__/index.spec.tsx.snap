// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`For should render fallback 1`] = `
[
  <div
    data-testid="loading"
  >
    Loading...
  </div>,
]
`;

exports[`For should render list all 1`] = `
[
  <div
    data-testid="item"
  >
    1
  </div>,
  <div
    data-testid="item"
  >
    2
  </div>,
  <div
    data-testid="item"
  >
    3
  </div>,
  <div
    data-testid="item"
  >
    4
  </div>,
  <div
    data-testid="item"
  >
    5
  </div>,
]
`;
